src/components/RuntimeManager.js

```diff
@@ -122,35 +122,20 @@ const AppErrorNotification = ({ app }) => {
   ])
 }
 
-//TODO: refactor into functional component
-export default class RuntimeManager extends PureComponent {
-  static propTypes = {
-    namespace: PropTypes.string.isRequired,
-    name: PropTypes.string.isRequired,
-    runtimes: PropTypes.array,
-    persistentDisks: PropTypes.array,
-    canCompute: PropTypes.bool.isRequired,
-    refreshRuntimes: PropTypes.func.isRequired,
-    workspace: PropTypes.object,
-    apps: PropTypes.array,
-    appDataDisks: PropTypes.array
-  }
-
-  render() {
-    return null
-  }
-
-  componentDidUpdate(prevProps) {
-    const { namespace, name, apps } = this.props
-    const prevRuntime = _.last(_.sortBy('auditInfo.createdDate', _.remove({ status: 'Deleting' }, prevProps.runtimes))) || {}
-    const runtime = this.getCurrentRuntime() || {}
+const RuntimeManager = ({ namespace, name, runtimes, apps }) => {
+  const prevRuntimes = usePrevious(runtimes)
+  const prevApps = usePrevious(apps)
+
+  useEffect(() => {
+    const runtime = getCurrentRuntime(runtimes) || {}
+    const prevRuntime = _.last(_.sortBy('auditInfo.createdDate', _.remove({ status: 'Deleting' }, prevRuntimes))) || {}
     const twoMonthsAgo = _.tap(d => d.setMonth(d.getMonth() - 2), new Date())
     const welderCutOff = new Date('2019-08-01')
-    const createdDate = new Date(runtime.createdDate)
-    const dateNotified = getDynamic(sessionStorage, `notifiedOutdatedRuntime${runtime.id}`) || {}
+    const createdDate = new Date(runtime?.createdDate)
```

Comment:
What has changed that now there may not be a current runtime?

Reply:
There could always have been no current runtime. The definition does use a default [here](https://github.com/DataBiosphere/terra-ui/pull/3345/files#diff-aabe65a220cb1e02fd8b0e1a38c532eb92df64bcc9b0e244801141b6330d1ca5R130), but I'd rather be defensive here since the data still comes from an API response and we have no way of enforcing contract adherence in the UI.

Reply:
I can't tell what the effect would be if the `runtime` ends up being undefined-- the variable access is defensive and wouldn't throw an error, but would the usage of those variables be robust to undefined? It might be worth thinking through that a bit.

Reply:
Essentially, if there is no runtime, there's no meaningful work this component has to do since there are no errors/notifications/status transitions that the user needs to be notified about

Reply:
In that case, it seems like it might be clearer to do something like:
```js
const runtime = getCurrentRuntime(runtimes)
if (!runtime) {
  return
}
```

vs have to deal with possibly undefined values all through the effect function.

Reply:
Ah, the same thing applies to apps though. So no meaningful work in terms of processing around that runtime's state if its undefined, but we can't return due to the possibility we need to notify on apps. Honestly, this whole component is a massive if/else tree that should be simplified into a Utils.cond. I really think something like that is necessary soon or we're talking about dressing up something gross here.